pyseeyou/locales.py

```diff
@@ -1,23 +1,53 @@
 from __future__ import unicode_literals
 
+LOCALE_SEP = "-"
 
-def get_cardinal_category(num_string, lang):
-    n, i, v, w, f, t = get_parts_of_num(num_string)
 
-    if lang in LOCALE_FUNCTIONS:
-        return LOCALE_FUNCTIONS[lang](n, i, v, w, f, t)
+# Ported from https://github.com/format-message/format-message/blob/master/packages/lookup-closest-locale/index.js#L4-L15
+def lookup_closest_locale(locale, available):
+    if isinstance(locale, str) and locale in available:
+        return locale
+
+    locales = []
+    if isinstance(locale, list):
+        locales = [] + locale
     else:
-        raise Exception('Unknown locale code.')
+        locales.append(locale)
+
+    for i in range(len(locales)):
+        current = locales[i].split(LOCALE_SEP)
+        while len(current) != 0:
+            candidate = LOCALE_SEP.join(current)
+            if candidate in available:
+                return candidate
+
+            current.pop()
+
+
+def get_cardinal_category(num_string, locale):
+    n, i, v, w, f, t = get_parts_of_num(num_string)
+
+    closest_locale = lookup_closest_locale(locale, cardinals)
+
+    if not closest_locale:
+        # Fallback to default locale
+        locale = "en"
+
+    cardinal_func = cardinals[closest_locale]
+
+    if not cardinal_func:
+        # Should be here different error message?
+        raise Exception(f'Locale "{locale}" not supported')
```

Comment:
Unfortunately, `f''` strings were only introduced in Python 3.6. This would break our compatibility with Python 2.7 and Python 3.5. Try using `'{}'.format()` instead?